### PR TITLE
[REG-1848] Gracefully handle IO errors during legacy input instrumentation

### PR DESCRIPTION
**Pull request by svolokh**
For various reasons, the initial attempt at instrumenting the legacy input API usage in the assemblies could fail (e.g. file in use by another job). 

This PR introduces graceful handling of these errors. If the initial instrumentation fails, a couple more attempts are scheduled with exponential backoff. Ultimately if all these attempts fail, the user is warned that the SDK's legacy input support may not work and the user should consider rebuilding the game code.

---

Find the pull request instructions [here](https://www.notion.so/regressiongg/Pull-Request-Process-0d57a7eb582a446983edfacafa406f1e?pvs=4)

Every reviewer and the owner of the PR should consider these points in their request (feel free to copy this checklist so you can fill it out yourself in the overall PR comment)

- [ ] The code is extensible and backward compatible
- [ ] New public interfaces are extensible and open to backward compatibility in the future
- [ ] If preparing to remove a field in the future (i.e. this PR removes an argument), the argument stays but is no longer functional, and attaches a deprecation warning. A linear task is also created to track this deletion task.
- [ ] Non-critical or potentially modifiable arguments are optional
- [ ] Breaking changes and the approach to handling them have been verified with the team (in the Linear task, design doc, or PR itself)
- [ ] The code is easy to read
- [ ] Unit tests are added for expected and edge cases
- [ ] Integration tests are added for expected and edge cases
- [ ] Functions and classes are documented
- [ ] Migrations for both up and down operations are completed
- [ ] A documentation PR is created and being reviewed for anything in this PR that requires knowledge to use
- [ ] Implications on other dependent code (i.e. sample games and sample bots) is considered, mentioned, and properly handled
- [ ] Style changes and other non-blocking changes are marked as non-blocking from reviewers

### CODE REVIEW

File: src/gg.regression.unity.bots/Editor/Scripts/RGLegacyInputUtility/RGLegacyInputInstrumentation.cs [hunk 1]
```diff
@@ -260,17 +259,77 @@ public static Assembly FindRGAssembly()
...
+        {
+            try
+            {
+                Assembly[] assemblies = CompilationPipeline.GetAssemblies(AssembliesType.Player);
+                Assembly rgAssembly = FindRGAssembly();
+                foreach (Assembly assembly in assemblies)
+                {
+                    InstrumentAssemblyIfNeeded(assembly.outputPath, rgAssembly);
+                }
+                return true;
```
Comment (RG-nAmKcAz) [inline src/gg.regression.unity.bots/Editor/Scripts/RGLegacyInputUtility/RGLegacyInputInstrumentation.cs:319]: do we need a 

`_numInstrumentationAttempts = 0;` 

code line here to reset this ?

Comment (RG-nAmKcAz) [inline src/gg.regression.unity.bots/Editor/Scripts/RGLegacyInputUtility/RGLegacyInputInstrumentation.cs:319]: do we need a `_numInstrumentationAttempts = 0;` code line here before the return to reset this ?

**Review by RG-nAmKcAz — APPROVED** [review]
One question about resetting the counters on a working instrumentation, but overall the concept seems good.

File: src/gg.regression.unity.bots/Editor/Scripts/RGLegacyInputUtility/RGLegacyInputInstrumentation.cs (hunk 1 above)
Comment (svolokh) [reply src/gg.regression.unity.bots/Editor/Scripts/RGLegacyInputUtility/RGLegacyInputInstrumentation.cs:320]: Technically it isn't needed since the loop can only ever happen once per static initialization, but for good measure I've added in a reset after the loop terminates:

https://github.com/Regression-Games/RGUnityBots/blob/3ab36906770a6cfa866c7bc823d03dee7fba3f81/src/gg.regression.unity.bots/Editor/Scripts/RGLegacyInputUtility/RGLegacyInputInstrumentation.cs#L303-L307

**Review by vontell — APPROVED** [review]
Great approach! Glad you implemented a backoff approach with limited retries.
